aws/table_aws_cognito_user_pool_identity_provider.go

```diff
@@ -0,0 +1,232 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cognitoidentityprovider"
+	"github.com/aws/aws-sdk-go-v2/service/cognitoidentityprovider/types"
+
+	cognitoidentityproviderv1 "github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsCognitoUserPoolIdentityProvider(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_cognito_user_pool_identity_provider",
```

Comment:
```suggestion
		Name:        "aws_cognito_identity_provider",
```

Based on the resource name in AWS, API names, etc., should this just be `aws_cognito_identity_provider`? Are there other identity provider resources in Cognito that we'd to differentiate between by adding `user_pool`?

Reply:
That's how I named it at first when I only looked at the [Amazon Cognito user pools](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-userpools-api-reference.html) [API reference](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/Welcome.html), but then I looked at the [Amazon Cognito identity pools (federated identities)](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-federatedidentities-api-reference.html) [API reference](https://docs.aws.amazon.com/cognitoidentity/latest/APIReference/Welcome.html) and figured there may be a confusion between these types:
* https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_IdentityProviderType.html
* https://docs.aws.amazon.com/cognitoidentity/latest/APIReference/API_CognitoIdentityProvider.html


An alternative may be to just embed these as an `identity_providers` columns in the `aws_cognito_user_pool` table.

Reply:
I think the chance for collision is low, considering that in the Cognito Federated Identities service, the [CognitoIdentityProvider](https://docs.aws.amazon.com/cognitoidentity/latest/APIReference/API_DescribeIdentityPool.html#CognitoIdentity-DescribeIdentityPool-response-CognitoIdentityProviders) is a property of an identity pool resource and is a reference to a user pool.

I like having Cognito User Pools identity providers as a separate table, in particular because the [ListIdentityProviders API](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_ListIdentityProviders.html) is paginated, so we'd generally recommend separating it out as per [our plugin release checklist](https://steampipe.io/docs/develop/plugin-release-checklist#data-ingestion).

However, I do see the chance for conflict between Cognito Identity and Cognito Sync resources, as there look to be specific ARNs for identities and identity pools in [Cognito Sync](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncognitosync.html#amazoncognitosync-resources-for-iam-policies). I'm not sure how these are used though as I couldn't see specific APIs for them in [Cognito Sync APIs](https://docs.aws.amazon.com/cognitosync/latest/APIReference/Welcome.html).

Overall, I found the Cognito services a bit confusing, as there appear to be 3 sets of endpoints.

For instance for us-east-2:
- cognito-idp.us-east-2.amazonaws.com (Amazon Cognito User Pools)
- cognito-identity.us-east-2.amazonaws.com (Amazon Cognito Identity Pools)
- cognito-sync.us-east-2.amazonaws.com (Amazon Cognito Sync)

Endpoints above taken from:
- https://docs.aws.amazon.com/general/latest/gr/cognito_identity.html
- https://docs.aws.amazon.com/general/latest/gr/cognito_sync.html

AWS seems to group Cognito User Pools and Identity Pools together in docs, though there are separate APIs and CLI commands for each of the endpoints services.

I don't think it's worth adding the words `idp`, `user_pool`, `identity_provider`, etc., after the word `cognito` by default to table names. So the table names `cognito_user_pool` and `cognito_identity_provider` LGTM. The only exception I think I'd make is for the Cognito Sync tables, e.g., `cognito_sync_dataset`.

Reply:
SGTM!

Renamed the table from `aws_cognito_user_pool_identity_provider` to `aws_cognito_identity_provider`.